flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/UserDefinedFunction.scala

```diff
@@ -40,6 +40,12 @@ abstract class UserDefinedFunction extends Serializable {
   @throws(classOf[Exception])
   def close(): Unit = {}
 
+  /**
+    * @return true iff a call to this function is guaranteed to always return
+    *         the same result given the same parameters; true is assumed by default
+    */
+  def isDeterministic: Boolean = true
```

Comment:
I suggest explain when user need to overwrite the method, what the impact if not overwrite the method.